### PR TITLE
port rest-json quickstart to camel-quarkus

### LINKED ISSUE #118: Decide how to maintain examples
Followup of the discussion in https://github.com/apache/camel-quarkus/issues/112 and https://github.com/apache/camel-quarkus/pull/115

What we agreed upon with @lburgazzoli :

* We need examples
* Integration tests may test aspects that have no didactic value for the app developers so keeping the distinction between itests and examples makes sense
* Examples should have tests

Open questions:

1. Where should the examples be hosted? `camel-quarkus` (preferred by @ppalaga ) or elsewhere?

2. Should the tests in example projects be run for every PR in `camel-quarkus`? 

* @ppalaga: yes definitely, to have a tight feedback and responsibility loop
* @lburgazzoli: no daily is enough, otherwise the CI would take long

### PR DESCRIPTION
**Pull request by lburgazzoli**
Fixes #118

### CODE REVIEW

File: examples/rest-json/pom.xml [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    </properties>
+
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>org.apache.camel.quarkus</groupId>
+                <artifactId>camel-quarkus-bom</artifactId>
+                <version>${project.version}</version>
+                <type>pom</type>
+                <scope>import</scope>
```
Comment (ppalaga) [inline examples/rest-json/pom.xml]: I guess we do not need both to import camel-quarkus-bom and have it as a parent?

File: examples/rest-json/pom.xml [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-platform-http</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.camel</groupId>
+            <artifactId>camel-jackson</artifactId>
+            <version>${camel.version}</version>
```
Comment (ppalaga) [inline examples/rest-json/pom.xml]: Shouldn't we manage camel-jackson in the BOM?

File: examples/rest-json/src/main/java/org/acme/rest/json/Routes.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+public class Routes extends RouteBuilder {
+    private final Set<Fruit> fruits;
+    private final Set<Legume> legumes;
+
+    public Routes() {
+        this.fruits =  Collections.newSetFromMap(Collections.synchronizedMap(new LinkedHashMap<>()));
+        this.fruits.add(new Fruit("Apple", "Winter fruit"));
+        this.fruits.add(new Fruit("Pineapple", "Tropical fruit"));
+
+        this.legumes = Collections.synchronizedSet(new LinkedHashSet<>());
```
Comment (ppalaga) [inline examples/rest-json/src/main/java/org/acme/rest/json/Routes.java]: Never heart of legumes and was happy to look it up in the dictionary :)

File: examples/rest-json/pom.xml (hunk 1 above)
Comment (lburgazzoli) [reply examples/rest-json/pom.xml]: done

File: examples/rest-json/pom.xml (hunk 2 above)
Comment (lburgazzoli) [reply examples/rest-json/pom.xml]: done

**Comment by lburgazzoli** [comment]
@ppalaga fixed (except the legumes thing)

File: examples/rest-json/src/main/resources/META-INF/resources/fruits.html [hunk 4]
```diff
@@ -0,0 +1,98 @@
+<!doctype html>
```
Comment (WillemJiang) [inline examples/rest-json/src/main/resources/META-INF/resources/fruits.html]: We need to add the apache license header on these two fruits.html.

**Pull request merged**